src/codeu/chat/server/Server.java

```diff
@@ -140,6 +156,10 @@ public void onMessage(InputStream in, OutputStream out) throws IOException {
 
         Serializers.INTEGER.write(out, NetworkCode.GET_ALL_CONVERSATIONS_RESPONSE);
         Serializers.collection(ConversationHeader.SERIALIZER).write(out, conversations);
+      if (type == NetworkCode.SERVER_INFO_REQUEST) {
```

Comment:
See my above comment - you only need to make that change once, so you can probably delete this section.